rosidl_generator_py/resource/_srv.py.em

```diff
@@ -1,7 +1,43 @@
 # generated from rosidl_generator_py/resource/_srv.py.em
 # generated code does not contain a copyright notice
 
-from @(package_name).msg._@(spec.srv_name)__request import @(spec.srv_name)_Request
-from @(package_name).msg._@(spec.srv_name)__response import @(spec.srv_name)_Response
-@(spec.srv_name)_Request  # unused
-@(spec.srv_name)_Response  # unused
+import logging
+import traceback
+
+
+class Metaclass(type):
+    """Metaclass of message '@(spec.srv_name)'."""
+    _TYPE_SUPPORT = None
+
+    @@classmethod
+    def __import_type_support__(cls):
+        __type_support_importable = False
+        try:
+            import rclpy
+            from rosidl_generator_py import import_type_support
+            __type_support_importable = True
+        except ImportError:
+            logger = logging.getLogger('rosidl_generator_py.@(spec.srv_name)')
+            logger.debug(
+                'Failed to import needed modules for type support:\n' + traceback.format_exc())
+
+        if __type_support_importable:
```

Comment:
I was not familiar with the try-except-else pattenr, fixed in https://github.com/ros2/rosidl/pull/171/commits/0d35a5ac70b82db44d380ab5a809f43fc3713a1f